pkg/webhooks/paddlepaddle/paddlepaddle_webhook.go

```diff
@@ -55,26 +56,33 @@ func (w Webhook) ValidateCreate(ctx context.Context, obj runtime.Object) (admiss
 	job := obj.(*trainingoperator.PaddleJob)
 	log := ctrl.LoggerFrom(ctx).WithName("paddlejob-webhook")
 	log.V(5).Info("Validating create", "paddleJob", klog.KObj(job))
-	return nil, validatePaddleJob(job).ToAggregate()
+	return nil, validatePaddleJob(nil, job).ToAggregate()
 }
 
-func (w Webhook) ValidateUpdate(ctx context.Context, _, newObj runtime.Object) (admission.Warnings, error) {
-	job := newObj.(*trainingoperator.PaddleJob)
+func (w Webhook) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (admission.Warnings, error) {
+	oldJob := oldObj.(*trainingoperator.PaddleJob)
+	newJob := newObj.(*trainingoperator.PaddleJob)
 	log := ctrl.LoggerFrom(ctx).WithName("paddlejob-webhook")
-	log.V(5).Info("Validating update", "paddleJob", klog.KObj(job))
-	return nil, validatePaddleJob(job).ToAggregate()
+	log.V(5).Info("Validating update", "paddleJob", klog.KObj(newJob))
+	return nil, validatePaddleJob(oldJob, newJob).ToAggregate()
 }
 
 func (w Webhook) ValidateDelete(context.Context, runtime.Object) (admission.Warnings, error) {
 	return nil, nil
 }
 
-func validatePaddleJob(job *trainingoperator.PaddleJob) field.ErrorList {
+func validatePaddleJob(oldJob, newJob *trainingoperator.PaddleJob) field.ErrorList {
 	var allErrs field.ErrorList
-	if errors := apimachineryvalidation.NameIsDNS1035Label(job.Name, false); len(errors) != 0 {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("metadata").Child("name"), job.Name, fmt.Sprintf("should match: %v", strings.Join(errors, ","))))
+	if errors := apimachineryvalidation.NameIsDNS1035Label(newJob.Name, false); len(errors) != 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("metadata").Child("name"), newJob.Name, fmt.Sprintf("should match: %v", strings.Join(errors, ","))))
 	}
-	allErrs = append(allErrs, validateSpec(job.Spec.PaddleReplicaSpecs)...)
+
+	if oldJob != nil {
+		allErrs = append(allErrs, util.ValidateRunPolicyUpdate(&oldJob.Spec.RunPolicy, &newJob.Spec.RunPolicy)...)
+	} else {
+		allErrs = append(allErrs, util.ValidateRunPolicy(&newJob.Spec.RunPolicy)...)
```

Comment:
This should be checked unconditionally. We want to run `ValidateRunPolicy` also in case of updates. This may not be needed for immutable fields, but in general this is a good practice in the core k8s.